server.go

```diff
@@ -137,13 +137,26 @@ func (s *Server) Serve(conn *net.UDPConn) {
 // bit we are relying on the low-level packet control channel info to
 // get a reliable source address, and those have different types and
 // the struct itself is not easily interface-ized or embedded.
+//
+// If control is nil for whatever reason (either things not being
+// implemented on a target OS or whatever other reason), the code will
+// fall back to getting the local address from the underlying USP
+// connection.  The address this returns is not always guaranteed to
+// be the dest IP of the underlying packet for complicated UDP
+// processing reasons that I don't fully understand myself because TCP
+// tends to always be accurate.
```

Comment:
That sounds reasonable.

FWIW, the control packet being nil only intermittently happens on darwin -- in the backtrace for the failed build (https://travis-ci.org/pin/tftp/jobs/217614000) the tests had gone through several ephemeral server/client pairs before things exploded, and on the mac I have access to things failed in different tests, although with more or less the same backtrace every time.